RoddyCore/src/de/dkfz/roddy/knowledge/files/BaseFile.groovy

```diff
@@ -93,16 +96,20 @@ public abstract class BaseFile<FS extends FileStageSettings> extends FileObject
         }
 
         public ConstructionHelperForGenericCreation(ExecutionContext context, ToolEntry creatingTool, String toolID, String slotID, String selectionTag, FileStageSettings fileStageSettings, JobResult jobResult) {
-            super(context, fileStageSettings, selectionTag, jobResult)
+            super(context, fileStageSettings, selectionTag, null, jobResult)
             this.slotID = slotID
             this.toolID = toolID
             this.creatingTool = creatingTool
         }
     }
 
     public static class ConstructionHelperForManualCreation extends ConstructionHelperForGenericCreation {
+        public ConstructionHelperForManualCreation(FileObject parentObject, List<FileObject> parentFiles, ToolEntry creatingTool, String toolID, String slotID, String selectionTag, String indexInFileGroup, FileStageSettings fileStageSettings, JobResult jobResult) {
```

Comment:
We can do that, however keep in mind, that custom types will make the lines even longer! I think it is less dangerous here because we don't use default parameters.

How about mandatorily named parameters? How is this done? Will it make lines longer?